vendor/engines/c2po/app/services/c2po/account_builder_extension.rb

```diff
@@ -0,0 +1,35 @@
+module C2po
+
+  # Includes AccountBuilder extensions for both CreditCardAccount and
+  # PurchaseOrderAccount account types.
+  module AccountBuilderExtension
+    extend ActiveSupport::Concern
+
+    # Dynamically called by `build_subclass` if the account_type is
+    # CreditCardAccount.
+    def build_credit_card_account(account, params)
+      account = build_credit_card_account_expires_at(account, params)
+      account
+    end
+
+    # Dynamically called by `build_subclass` if the account_type is
+    # PurchaseOrderAccount.
+    def build_purchase_order_account(account, params)
+      account = build_purchase_order_account_expires_at(account, params)
+      account
+    end
+
+    # Sets `expires_at` for CreditCardAccount only.
+    def build_credit_card_account_expires_at(account, params)
+      account.expires_at = Date.civil(account.expiration_year.to_i, account.expiration_month.to_i).end_of_month.end_of_day
+      acount
+    end
+
+    # Sets `expires_at` for PurchaseOrderAccount only.
+    def build_purchase_order_account_expires_at(account, params)
+      account.expires_at = parse_usa_date(account.expires_at).try(:end_of_day)
+      account
+    end
```

Comment:
I __think__ that I captured the `expires_at` setters for both credit card and purchase orders, but I'm not 100% sure.  Please review.